### PR TITLE
rocksdb: sync with upstream

### PR DESCRIPTION
**Pull request by tchaikov**
to avoid warning with glibc 2.25, see
https://github.com/facebook/rocksdb/pull/2208 and
https://github.com/ceph/rocksdb/pull/17

Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
```
Agent went offline during the build
```
retest this please.

**Comment by tchaikov** [comment]
retest this please.

**Comment by smithfarm** [comment]
@tchaikov This fixes one build error, and causes another one (in ppc64le):

```
[  370s] c++: error: unrecognized command line option '-msse4.2'
```

**Comment by smithfarm** [comment]
Introduced by https://github.com/facebook/rocksdb/commit/bc397325

**Comment by smithfarm** [comment]
Test-building this now on POWER8

**Comment by smithfarm** [comment]
@tchaikov This doesn't quite do it

```
[  308s] CMake Warning:
[  308s]   Manually-specified variables were not used by the project:
[  308s] 
[  308s]     WITH_MSSE42
```

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -768,6 +768,8 @@ if (NOT WITH_SYSTEM_ROCKSDB)
   # with SSE 4.2. For details refer to rocksdb/util/crc32c.cc.
   if (HAVE_INTEL_SSE4_2)
     list(APPEND ROCKSDB_CMAKE_ARGS -DCMAKE_CXX_FLAGS=${SIMD_COMPILE_FLAGS})
+  else()
+    list(APPEND ROCKSDB_CMAKE_ARGS -DWITH_MSSE42=OFF)
```
Comment (smithfarm) [inline src/CMakeLists.txt]: @tchaikov - sorry, typo, please s/WITH_MSSE42/WITH_SSE42/

Comment (smithfarm) [inline src/CMakeLists.txt]: You might as well wait until I finish testing. . .

**Comment by tchaikov** [comment]
@smithfarm does it build in your testbed?

**Comment by smithfarm** [comment]
x86_64 build is green, too (in OBS)

**Comment by yuriw** [comment]
http://pulpito.ceph.com/yuriw-2017-04-27_22:07:22-rados-wip-yuri-testing_2017_4_28_2-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-04-27_22:07:22-rados-wip-yuri-testing_2017_4_28_2-distro-basic-smithi/

**Pull request merged**